### DISCUSSION TITLE
Couple random general questions

### DISCUSSION THREAD

**Discussion by xstrex** [ttscoff/bunch-docs#272] [Q&A] [answered]
Still new to working with Bunch, but like it's functionality & versatility. Have a couple of general questions I couldn't find answers to yet. 

- ~~How does Bunch behave with bartender, any gotcha's or issues?~~
- Does Bunch support, or have compatibility with external monitors, and other peripherals - more for detection, and less for window placement. Like running a Bunch when a laptop is docked, and a different one when it's undocked? (think toggling internal keyboard back lighting when an external keyboard is connected)
- Any good or recommended guides for writing AppleScript? (familiarity with several other scripting languages) 
- Is AppleScript the best tool to use to control applications that don't have (or publish) command line arguments or APIs? 
- Any way to trigger or launch a Bunch other than using the Bunch menu? (I've seen the raycast integration but it doesn't seem to support launching a bunch, which seems odd. Any planned support for launching a bunch from Shortcuts)

Cheers

**Comment by ttscoff** [comment, marked as answer]
> - How does Bunch behave with bartender, any gotcha's or issues?

Using the hotkey to trigger the menu while Bunch is hidden by Bartender will usually not work. To have hotkey access to Bunches, make Bunch always visible.

> - Does Bunch support, or have compatibility with external monitors, and other peripherals - more for detection, and less for window placement. Like running a Bunch when a laptop is docked, and a different one when it's undocked? (think toggling internal keyboard back lighting when an external keyboard is connected)

No, Bunch has no current support for using peripheral detection as logic for actions. Might consider it for the future, though.

> - Any good or recommended guides for writing AppleScript? (familiarity with several other scripting languages)

There are thousands of resources. The thing with AppleScript is every app defines its own dictionary, so an AppleScript that accomplishes a task in one app has little likelihood of working in another app. You basically have to google "Do THING in APP_NAME applescript" every time. The resources are there, but there's no canonical reference.

> - Is AppleScript the best tool to use to control applications that don't have (or publish) command line arguments or APIs?

Yes, though if you can also perform an action with Automator, Shortcuts, or an third-party tool like BetterTouchTool or Keyboard Maestro, all of those are easy to integrate directly into a Bunch.

> - Any way to trigger or launch a Bunch other than using the Bunch menu? (I've seen the raycast integration but it doesn't seem to support launching a bunch, which seems odd. Any planned support for launching a bunch from Shortcuts)

There's a URL handler for triggering Bunches. I would assume the Raycast extension supports this, I know the LaunchBar and Alfred ones do.

https://bunchapp.co/docs/integration/url-handler/

You can also trigger bunches with AppleScript, so any tool that can run a single line of AppleScript can also open, close, and toggle Bunches.

https://bunchapp.co/docs/integration/applescript/

-Brett
